include/libopencm3/stm32/h7/rcc.h

```diff
@@ -455,13 +468,13 @@ struct rcc_pll_config {
     uint8_t divq;                   /**< Post divider for PLLQ clock. */
     uint8_t divr;                   /**< Post divider for PLLR clock. */
   } pll1, pll2, pll3;               /**< PLL1-PLL3 configurations. */
-  uint8_t core_pre;                 /**< Core prescaler  note: domain 1. */
-  uint8_t hpre;                     /**< HCLK3 prescaler note: domain 1. */
-  uint8_t ppre1;                    /**< APB1 Peripheral prescaler note: domain 2. */
-  uint8_t ppre2;                    /**< APB2 Peripheral prescaler note: domain 2. */
-  uint8_t ppre3;                    /**< APB3 Peripheral prescaler note: domain 1. */
-  uint8_t ppre4;                    /**< APB4 Peripheral prescaler note: domain 3. */
-  uint8_t flash_waitstates;         /**< Latency Value to set for flahs. */
+  enum rcc_core_pre core_pre;       /**< Core prescaler. */
+  enum rcc_domain1_hpre hpre;       /**< HCLK3 prescaler. */
+  enum rcc_domain2_ppre ppre1;      /**< APB1 Peripheral prescaler. */
+  enum rcc_domain2_ppre ppre2;      /**< APB2 Peripheral prescaler. */
+  enum rcc_domain1_ppre ppre3;      /**< APB3 Peripheral prescaler. */
+  enum rcc_domain3_ppre ppre4;      /**< APB4 Peripheral prescaler. */
+  uint8_t flash_waitstates;         /**< Latency Value to set for flash. FLASH_ACR_LATENCY_xWS value. */
```

Comment:
this is out of scope, and aren't the flash wait states pure numeric anyway?

Reply:
These shift value for this macro is zero, so it works either way, but I agree it's sorta outside of the scope of changing the rcc regs to enums